cfn-resources/global-cluster-config/cmd/resource/resource.go

```diff
@@ -246,53 +226,58 @@ func validateModel(fields []string, model *Model) *handler.ProgressEvent {
+		addManagedNamespace := &admin.DeleteManagedNamespaceApiParams{
+			Collection:  m.Collection,
+			Db:          m.Db,
+			ClusterName: clusterName,
+			GroupId:     projectID,
 		}
-		addManagedNamespace.IsCustomShardKeyHashed = aws.Bool(*m.IsCustomShardKeyHashed)
-		addManagedNamespace.IsShardKeyUnique = aws.Bool(*m.IsShardKeyUnique)
-		_, _, err := conn.GlobalClusters.DeleteManagedNamespace(ctx, projectID, clusterName, addManagedNamespace)
+
+		_, _, err := conn.AtlasV2.GlobalClustersApi.DeleteManagedNamespaceWithParams(ctx, addManagedNamespace).Execute()
 		if err != nil {
 			_, _ = logger.Warnf("error while removing namespace:%+v", err)
 		}
 	}
 }
 
-func addManagedNamespaces(ctx context.Context, client *mongodbatlas.Client, nameSpaces []ManagedNamespace, projectID, clusterName string) {
+func newCustomZoneMappings(currentModel *Model) *admin.CustomZoneMappings {
+	return &admin.CustomZoneMappings{
+		CustomZoneMappings: modelToCustomZoneMappings(currentModel.CustomZoneMappings),
+	}
+}
+
+func createManagedNamespaces(ctx context.Context, client *util.MongoDBClient, nameSpaces []ManagedNamespace, projectID, clusterName string) error {
 	for _, mn := range nameSpaces {
-		addManagedNamespace := &mongodbatlas.ManagedNamespace{
-			Collection:     cast.ToString(mn.Collection),
-			Db:             cast.ToString(mn.Db),
-			CustomShardKey: cast.ToString(mn.CustomShardKey),
+		addManagedNamespace := &admin.ManagedNamespace{
+			Collection:     mn.Collection,
+			Db:             mn.Db,
+			CustomShardKey: mn.CustomShardKey,
 		}
 		addManagedNamespace.IsCustomShardKeyHashed = mn.IsCustomShardKeyHashed
 		addManagedNamespace.IsShardKeyUnique = mn.IsShardKeyUnique
-		_, _, err := client.GlobalClusters.AddManagedNamespace(ctx, projectID, clusterName, addManagedNamespace)
+		_, _, err := client.AtlasV2.GlobalClustersApi.CreateManagedNamespace(ctx, projectID, clusterName, addManagedNamespace).Execute()
 		if err != nil {
-			_, _ = logger.Warnf("error while adding namespace:%+v", err)
+			return err
 		}
 	}
-}
 
-func modelToCustomZoneMapping(tfMap ZoneMapping) *mongodbatlas.CustomZoneMapping {
-	return &mongodbatlas.CustomZoneMapping{
-		Location: cast.ToString(tfMap.Location),
-		Zone:     cast.ToString(tfMap.Zone),
-	}
+	return nil
 }
 
-func modelToCustomZoneMappings(tfList []ZoneMapping) []mongodbatlas.CustomZoneMapping {
-	apiObjects := make([]mongodbatlas.CustomZoneMapping, len(tfList))
+func modelToCustomZoneMappings(tfList []ZoneMapping) []admin.ZoneMapping {
+	apiObjects := make([]admin.ZoneMapping, len(tfList))
 	for i, tfMapRaw := range tfList {
-		if tfMapRaw == (ZoneMapping{}) || tfMapRaw.Location == nil || tfMapRaw.Zone == nil {
+		if !util.IsStringPresent(tfMapRaw.Location) || !util.IsStringPresent(tfMapRaw.Zone) {
 			continue
 		}
-		apiObject := modelToCustomZoneMapping(tfMapRaw)
-		apiObjects[i] = *apiObject
+
+		apiObjects[i] = admin.ZoneMapping{
+			Location: *tfMapRaw.Location,
+			Zone:     *tfMapRaw.Zone,
+		}
 	}
```

Comment:
this is ok.  Another approach would be, maybe a bit easier to read:

if util.IsStringPresent(tfMapRaw.Location) && util.IsStringPresent(tfMapRaw.Zone) {
		apiObjects[i] = admin.ZoneMapping{
			Location: *tfMapRaw.Location,
			Zone:     *tfMapRaw.Zone,
		}
}

Reply:
updated

Reply:
ah, ok